lib/galaxy/webapps/galaxy/api/group_roles.py

```diff
@@ -1,19 +1,92 @@
 """
 API operations on Group objects.
 """
+
 import logging
 
+from fastapi import Path
+from routes.util import url_for
+
 from galaxy.managers.context import ProvidesAppContext
 from galaxy.managers.group_roles import GroupRolesManager
 from galaxy.schema.fields import EncodedDatabaseIdField
+from galaxy.schema.schema import GroupRoleListModel, GroupRoleModel
 from galaxy.web import (
     expose_api,
     require_admin,
 )
-from . import BaseGalaxyAPIController, depends
+from galaxy.webapps.galaxy.api.common import fastapi_deprecation_message
+from . import (
+    BaseGalaxyAPIController,
+    depends,
+    DependsOnTrans,
+    Router
+)
+
 
 log = logging.getLogger(__name__)
 
+router = Router(tags=['group_roles'])
+
+GroupIDParam: EncodedDatabaseIdField = Path(
+    ...,
+    title='GroupID',
+    description='The ID of the group'
+)
+
+RoleIDParam: EncodedDatabaseIdField = Path(
+    ...,
+    title='RoleID',
+    description='The ID of the role'
+)
+
+
+def group_role_to_model(trans, encoded_group_id, role):
+    encoded_role_id = trans.security.encode_id(role.id)
+    try:
+        url = url_for('group_role', group_id=encoded_group_id, id=encoded_role_id)
+    except AttributeError:
+        url = fastapi_deprecation_message()
+
+    return GroupRoleModel(
+        id=encoded_role_id,
+        name=role.name,
+        url=url
+    )
+
+
+@router.cbv
+class FastAPIGroupRoles:
+    manager: GroupRolesManager = depends(GroupRolesManager)
+
+    @router.get('/api/groups/{group_id}/roles',
+                require_admin=True,
+                summary='Displays a collection (list) of groups.')
+    def index(self, trans: ProvidesAppContext = DependsOnTrans, group_id: EncodedDatabaseIdField = GroupIDParam) -> GroupRoleListModel:
+        group_roles = self.manager.index(trans, group_id)
+        return GroupRoleListModel(__root__=[group_role_to_model(trans, group_id, gr.role) for gr in group_roles])
+
+    @router.get('/api/groups/{group_id}/roles/{role_id}',
+                require_admin=True,
+                summary='Displays information about a group role.')
+    def show(self, trans: ProvidesAppContext = DependsOnTrans, group_id: EncodedDatabaseIdField = GroupIDParam, role_id: EncodedDatabaseIdField = RoleIDParam) -> GroupRoleModel:
```

Comment:
This is only a suggestion. In Galaxy we don't have a formal rule for maximum line length. I try to keep mine within 100, whenever possible. The reason for that is that even with the widest monitor it is still very convenient to have multiple files, or splits, open at the same time. Exceedingly long lines get in the way. My suggestion would be to break up such long lines.